### PR TITLE
Winter 2021 Modifications

### PR DESCRIPTION
**Pull request by teiknsmith**
Some smaller modifications that have been requested for the new semester by myself or Riley.

- Tab autocomplete in filepath entry
- New output format for Submission Search
    - Cleaned up regex logic I inserted previously in a quick fix (it wasn't super great before)
- Class Section Groups
    - New attribute in class section
    - In class section selection in grade puller, (de)selection of a whole group with a lead toggle

### CODE REVIEW

File: zygrader/ui/layers.py [hunk 1]
```diff
@@ -316,6 +316,20 @@ def build(self):
             prompt += [f"[Invalid {TYPE_STR}]"]
         self.component.set_message(prompt)
 
+    def __autocomplete(self):
+        path = os.path.expanduser(self.component.text)
+        directory, partial_name = os.path.split(path)
```
Comment (natecraddock) [inline zygrader/ui/layers.py:321]: There is a crash in autocomplete if the path is invalid

```suggestion
        directory, partial_name = os.path.split(path)
        if not os.path.exists(directory):
            return
```

File: zygrader/data/model.py [hunk 2]
```diff
@@ -126,14 +133,17 @@ def from_json(cls, section_json):
         default_due_time = (datetime.datetime.strptime(
             default_due_time_str,
             ClassSection.DUE_TIME_STORAGE_FORMAT).astimezone(tz=None).time())
-        return ClassSection(section_number, default_due_time)
+        # Section Groups were added later, so do a safer get to avoid KeyError
+        section_group = section_json.get("section_group")
```
Comment (natecraddock) [inline zygrader/data/model.py:137]: Would you want to add a default here, or is `None` the best default? For example, you could do `default="Default Section Group"` as the second parameter.

**Review by natecraddock — COMMENTED** [review]
Looks good to me! Just a small thing I noticed, and a crash. Left those comments in the code.

File: zygrader/data/model.py (hunk 2 above)
Comment (teiknsmith) [reply zygrader/data/model.py:137]: I think None works well. The constructor handles this case, and I'd rather the special string be in just one location.

**Pull request merged**